quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java

```diff
@@ -273,8 +274,32 @@ void configureConfigSources(BuildProducer<StaticInitConfigSourceProviderBuildIte
     }
 
     @BuildStep(onlyIf = IsIntegrationTest.class)
-    void prepareTestEnvironment(BuildProducer<StaticInitConfigSourceProviderBuildItem> configSources) {
+    void prepareTestEnvironment(BuildProducer<StaticInitConfigSourceProviderBuildItem> configSources, DevServicesDatasourceResultBuildItem dbConfig) {
         configSources.produce(new StaticInitConfigSourceProviderBuildItem("org.keycloak.quarkus.runtime.configuration.test.TestKeycloakConfigSourceProvider"));
+
+        // we do not enable dev services by default and the DevServicesDatasourceResultBuildItem might not be available when discovering build steps
+        // Quarkus seems to allow that when the DevServicesDatasourceResultBuildItem is not the only parameter to the build step
+        // this might be too sensitive and break if Quarkus changes the behavior
+        if (dbConfig != null && dbConfig.getDefaultDatasource() != null) {
+            Map<String, String> configProperties = dbConfig.getDefaultDatasource().getConfigProperties();
+
+            for (Entry<String, String> dbConfigProperty : configProperties.entrySet()) {
+                PropertyMapper mapper = PropertyMappers.getMapper(dbConfigProperty.getKey());
+
+                if (mapper == null) {
+                    continue;
+                }
+
+                String kcProperty = mapper.getFrom();
+
+                if (kcProperty.endsWith("db")) {
```

Comment:
I'm not sure about this check, can you explain?

Reply:
We don't set `db` option because it holds the transformed value (quarkus specific) rather than the keycloak db alias.